src/Core/Services/ExecutionHelper.cs

```diff
@@ -0,0 +1,517 @@
+        /// <param name="variables">the request context variable values needed to resolve value nodes represented as variables</param>
+        public static object? ExtractValueFromIValueNode(
+            IValueNode value,
+            IInputField argumentSchema,
+            IVariableValueCollection variables)
+        {
+            // extract value from the variable if the IValueNode is a variable
+            if (value.Kind == SyntaxKind.Variable)
+            {
+                string variableName = ((VariableNode)value).Name.Value;
+                IValueNode? variableValue = variables.GetVariable<IValueNode>(variableName);
+
+                if (variableValue is null)
+                {
+                    return null;
+                }
+
+                return ExtractValueFromIValueNode(variableValue, argumentSchema, variables);
+            }
+
+            if (value is NullValueNode)
+            {
+                return null;
+            }
+
+            return argumentSchema.Type.TypeName().Value switch
+            {
+                SupportedHotChocolateTypes.BYTE_TYPE => ((IntValueNode)value).ToByte(),
+                SupportedHotChocolateTypes.SHORT_TYPE => ((IntValueNode)value).ToInt16(),
+                SupportedHotChocolateTypes.INT_TYPE => ((IntValueNode)value).ToInt32(),
+                SupportedHotChocolateTypes.LONG_TYPE => ((IntValueNode)value).ToInt64(),
+                SupportedHotChocolateTypes.SINGLE_TYPE => ((FloatValueNode)value).ToSingle(),
+                SupportedHotChocolateTypes.FLOAT_TYPE => ((FloatValueNode)value).ToDouble(),
+                SupportedHotChocolateTypes.DECIMAL_TYPE => ((FloatValueNode)value).ToDecimal(),
+                SupportedHotChocolateTypes.UUID_TYPE => Guid.TryParse(value.Value!.ToString(), out Guid guidValue) ? guidValue : value.Value,
+                _ => value.Value
+            };
+        }
+
+        /// <summary>
+        /// Extract parameters from the schema and the actual instance (query) of the field
+        /// Extracts default parameter values from the schema or null if no default
+        /// Overrides default values with actual values of parameters provided
+        /// Key: (string) argument field name
+        /// Value: (object) argument value
+        /// </summary>
+        public static IDictionary<string, object?> GetParametersFromSchemaAndQueryFields(
+            IObjectField schema,
+            FieldNode query,
+            IVariableValueCollection variables)
+        {
+            IDictionary<string, object?> parameters = new Dictionary<string, object?>();
+
+            // Fill the parameters dictionary with the default argument values
+            IFieldCollection<IInputField> argumentSchemas = schema.Arguments;
+
+            foreach (IInputField argument in argumentSchemas)
+            {
+                if (argument.DefaultValue != null)
```

Comment:
Correct. I have updated this function with more comments and examples of what is collected when a query is processed. This particular section attempts to resolve default values from default query arguments like `first`, `after`, `filter`, and `orderBy` but I can't identify any circumstances where we set a default. 

I'm not going to remove this preexisting code because i am unsure if that would be a breaking change. Currently, until we identify a specific case where defaults are present, this loop is a no-op.

Reply:
found a scenario in an existing test. will document and keep this code.

Reply:
mutation that executes a stored procedure where a default stored procedure parameter is defined in the runtime config.